FreeRTOS-Plus/Demo/MQTT_Mutual_Auth_Qemu_mps2/source/startup.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * FreeRTOS V202212.00
+ * Copyright (C) 2023 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "CMSIS/CMSDK_CM3.h"
+#include "CMSIS/core_cm3.h"
+
+extern void vPortSVCHandler( void );
+extern void xPortPendSVHandler( void );
+extern void xPortSysTickHandler( void );
+extern void uart_init();
+extern int main();
+
+void __attribute__((weak)) EthernetISR (void);
+
+extern uint32_t _estack, _sidata, _sdata, _edata, _sbss, _ebss;
+
+/* Prevent optimization so gcc does not replace code with memcpy */
+__attribute__((optimize("O0")))
+__attribute__((naked))
+void Reset_Handler(void)
+ {
+    // set stack pointer
+    __asm volatile ("ldr r0, =_estack");
+    __asm volatile ("mov sp, r0");
+    // copy .data section from flash to RAM
+    for (uint32_t *src = &_sidata, *dest = &_sdata; dest < &_edata;)
+    {
+        *dest++ = *src++;
+    }
```

Comment:
Why not use a `memcpy` here?

Reply:
We specifically want to prevent the usage of `memcpy` because this is startup code that runs when the board starts up (see comment on top of the function name).

Reply:
My bad, makes complete sense in the context. Thank you.